Source/WebKit/UIProcess/PageLoadState.cpp

```diff
@@ -103,7 +103,11 @@ void PageLoadState::commitChanges()
     bool negotiatedLegacyTLSChanged = m_committedState.negotiatedLegacyTLS != m_uncommittedState.negotiatedLegacyTLS;
     bool estimatedProgressChanged = estimatedProgress(m_committedState) != estimatedProgress(m_uncommittedState);
     bool networkRequestsInProgressChanged = m_committedState.networkRequestsInProgress != m_uncommittedState.networkRequestsInProgress;
-    bool certificateInfoChanged = m_committedState.certificateInfo != m_uncommittedState.certificateInfo;
+#if PLATFORM(COCOA)
+    bool certificateInfoChanged = m_committedState.certificateInfo.trust() != m_uncommittedState.certificateInfo.trust();
+#else
+    bool certificateInfoChanged = false;
+#endif
```

Comment:
Why is this cocoa specific now?

Reply:
yeah, this isn't great.  We were doing pointer comparison of the WebCertificateInfo before, which was not great, and we need a replacement.  We need WebCore::CertificateInfo to have an operator== on all platforms

Reply:
For soup that would be easy, something like this (untested):

```
bool CertificateInfo::operator==(const CertificateInfo& other)
{
    return m_tlsErrors == other.m_tlsErrors && g_tls_certificate_is_same(m_certificate.get(), other->m_certificate.get());
}
```

Reply:
And looks like soup is the only platform that doesn't already have the operator==, so that should do it.